src/pages/studyView/studyPageHeader/StudyPageHeader.tsx

```diff
@@ -48,42 +48,44 @@ export default class StudyPageHeader extends React.Component<
                     <RightPanel store={this.props.store} />
                 </div>
 
-                {this.props.store.clinicalAttributeIdToDataType.isComplete && (
-                    <UserSelections
-                        filter={this.props.store.userSelections}
-                        comparisonGroupSelection={
-                            this.props.store.filterComparisonGroups
-                        }
-                        numberOfSelectedSamplesInCustomSelection={
-                            this.props.store
-                                .numberOfSelectedSamplesInCustomSelection
-                        }
-                        customChartsFilter={this.props.store.customChartFilterSet.toJS()}
-                        attributesMetaSet={
-                            this.props.store.chartMetaSetWithChartType
-                        }
-                        clinicalAttributeIdToDataType={
-                            this.props.store.clinicalAttributeIdToDataType
-                                .result!
-                        }
-                        updateClinicalDataFilterByValues={
-                            this.props.store.updateClinicalDataFilterByValues
-                        }
-                        updateCustomChartFilter={
-                            this.props.store.setCustomChartFilters
-                        }
-                        removeGeneFilter={this.props.store.removeGeneFilter}
-                        removeCustomSelectionFilter={
-                            this.props.store.removeCustomSelectFilter
-                        }
-                        removeComparisonGroupSelectionFilter={
-                            this.props.store
-                                .removeComparisonGroupSelectionFilter
-                        }
-                        onBookmarkClick={this.props.onBookmarkClick}
-                        clearAllFilters={this.props.store.clearAllFilters}
-                    />
-                )}
+                {this.props.store.clinicalAttributeIdToDataType.isComplete &&
+                    this.props.store.bookmarkUrl.isComplete && (
```

Comment:
Add this line to ensure we have bookmarkUrl data ready, then we render the bookmark button.

Reply:
I think we should create bookmarkUrl only when the button is clicked. or it would create bitly url for each and every chart filter

Reply:
Thanks @kalletlak , good catch, I will try to update this.

Reply:
agreed